packages/pieces/facebook-pages/src/lib/common/index.ts

```diff
@@ -0,0 +1,163 @@
+import { OAuth2PropertyValue, Property } from "@activepieces/pieces-framework";
+import { HttpRequest, HttpMethod, httpClient } from "@activepieces/pieces-common";
+
+export const facebookPagesCommon = {
+    baseUrl: 'https://graph.facebook.com/v17.0',
+
+    authentication: Property.OAuth2({
+        displayName: 'Authentication',
+        description: '',
+        authUrl: "https://graph.facebook.com/oauth/authorize?",
+        tokenUrl: "https://graph.facebook.com/oauth/access_token",
+        required: true,
+        scope: ['pages_show_list', 'pages_manage_posts', 'pages_read_engagement'],
+    }),
+
+    page: Property.Dropdown({
+        displayName: 'Page',
+        required: true,
+        refreshers: ['authentication'],
+        options: async (props) => {
+            if (!props['authentication']) {
+                return {
+                    disabled: true,
+                    options: [],
+                    placeholder: 'Connect your account'
+                }
+            }
+
+            try {
+                const authProp: OAuth2PropertyValue = props['authentication'] as OAuth2PropertyValue;
```

Comment:
You could use the getAccessTokenOrFail helper to throw an exception in case it's missing